test/test_torch.py

```diff
@@ -6293,6 +6293,125 @@ def test_cat_different_devices(self, devices):
                                     "input tensors must be on the same device"):
             torch.cat((cpu, cuda0))
 
+    def test_block_diag(self, device):
```

Comment:
We should add tests that assert an error message is raised in any of our edge cases:
1. If any of the input tensor's dimensions is >= 3
2. If any of the input tensors are on different devices. The test case can just be creating a cpu tensor and a cuda tensor and calling block_diag

```
with self.assertRaisesRegex(RuntimeError, regex)
```
should be helpful here.